README.md

```diff
@@ -142,46 +152,60 @@ There is no strict limit over what you can do with a `Matrix`,
 
-The engine applies the given transformation, and any minZoom and maxZoom constraints.
-
-If, after this process, the content is bigger than the container, the engine will also apply a
-translation according to the given transformation gravity.
+After transformation is applied, the transformation gravity will reposition the content with
+the specified value. Supported values are most of the Android `Gravity` flags, plus `TRANSFORMATION_GRAVITY_AUTO`.
 
 |Transformation Gravity|Description|
 |----------------------|-----------|
-|`top`|If the content is taller than the view, translate it so that we see the top part.|
-|`bottom`|If the content is taller than the view, translate it so that we see the bottom part.|
-|`left`|If the content is wider than the view, translate it so that we see the left part.|
-|`right`|If the content is wider than the view, translate it so that we see the right part.|
+|`top`, ...|The content is panned so that its *top* side matches teh container *top* side. Same for other values.|
+|`auto` (default)|The transformation gravity is taken from the engine [alignment](#alignment), defaults to `center` on both axes.|
 
-If, after this process, the content is smaller than the container, note that the current
-[Smaller Policy](#smaller-policy) applies.
+**Note: after transformation and gravity are applied, the engine will apply - as always - all the active constraints,
+including minZoom, maxZoom, alignment. This means that the final position might be slightly (or completely) different.**
 
-Note: you can always trigger a new transformation to be applied by using the `setContentSize` or `setContainerSize` APIs.
+For example, when `maxZoom == 1`, is forced to not be any larger than the container. This means that
+a `centerCrop` transformation will not work (will act just like a `centerInside`).
 
-#### Smaller policy
+#### Alignment
 
-You can control how the content will be positioned when it is smaller than the container through 
-the `setSmallerPolicy(int)` method or by using the `smallerPolicy` XML attribute of `ZoomLayout` and `ZoomImageView`.
-By default, content is always centered when it is smaller than its container.
+You can force the content position with respect to the container using the `setAlignment(int)` method
+or the `alignment` XML flag of `ZoomLayout` and `ZoomImageView`. 
+The default value is `Alignment.CENTER` which will center the content on both directions.
 
-|Policy|Description|
-|------|-----------|
-|`center`|The content will be centered within the container.|
-|`fromTransformation`|The content will respect the gravity parameter of the transformation (which defaults to `Gravity.CENTER` as well).|
-|`none`|The content is free to be moved around inside the container bounds.|
+**Note: alignment does not make sense when content is larger than the container, because forcing an 
+alignment (e.g. left) would mean making part of the content unreachable (e.g. the right part).**
```

Comment:
[...] so it is only applied when the content is smaller than the container (e.g. `zoom < 1`)

Reply:
I will address all these in a while. But wanted to say that `zoom < 1` does not necessarily mean that content is smaller than container. It does only with CENTER_CROP or CENTER_INSIDE transformations and even then, it depends on the axis.

Just saying because I have seen this check (zoom < 1F) in the pivot function, and I don't know if it's correct or not. I'm tempted to say that it's not, and I also wonder if that should be changed according to the alignment? Not sure.

Reply:
Hmm yes that makes sense. I guess I was lucky in my testing.

Yes I think so too. As i mentioned in the comment there I initially wanted to base it on the transformationGravity but as the alignment will be the new thing after this PR it should be based on it instead.

Reply:
Well the function decides what pivot point should be used for the zoom animation based on what corner of the screen is currently overscrolled (or will be, with the target zoom applied). I did this because applying the pan fix and the zoom fix at the same time in an animation when overscrolled while zoomed in a bit didn't result in the expected target position and/or a funky animation. Implementing this was a bit of trial and error tbh and I think this method is only applicable when the content is zoomed in meaning that **at most** two sides of the content are overscrolled/have visible gray areas. Because of that I added the `zoom < 1` condition which just takes the center of the screen as the pivot point when the view has more than two gray areas around it (which doesn't seem to be the correct way of figuring that out). My current guess would be that this should be based on the alignment instead of just using the center when there are more than two gray areas but this whole method (and it's purpose) may require a complete rethinking.